### PR TITLE
Remove a section about server side clones

### PR DESCRIPTION
**Pull request by berkerpeksag**
We can now create branches on our forks.

### CODE REVIEW

File: committing.rst [hunk 1]
```diff
@@ -521,93 +521,6 @@ specific patch for ``3.6``.
...
-for collaboration with folks that don't already have CPython commit rights),
-Bitbucket_ also maintain an `up to date clone`_ of the main ``cpython``
-repository that can be used as the basis for a new clone or patch queue.
-
-.. _named branch: https://www.mercurial-scm.org/wiki/NamedBranches
-.. _Bitbucket: http://www.bitbucket.org
-.. _up to date clone: https://bitbucket.org/mirror/cpython/overview
-
-
 Uploading a patch for review
```
Comment (willingc) [inline committing.rst]: @berkerpeksag Looks good. 👍  Do you want to bump this section title?

```
Uploading a patch for review
-----------------------------
```

Comment (berkerpeksag) [inline committing.rst]: Oops, good catch, thanks :) I think we can remove it too.

Comment (berkerpeksag) [inline committing.rst]: Just removed the section, thanks.

**Comment by willingc** [comment]
Thanks @berkerpeksag 🐍

**Pull request merged**